### PR TITLE
(WIP) Update pokey.c with latest from MAME

### PR DESCRIPTION
**Pull request by jprjr**
A few notes -

MuteMask is untested

I tried to follow the C++ version as closely as possible and adopted a
few conventions:

C++ class pokey_device -> C struct pokey_device

C++ class pokey_channel -> C struct pokey_channel

Methods were renamed to include the classname, ie:

C++ method pokey_device::vol_init -> pokey_device_vol_init

device_start was basically moved into device_reset, pokey_device_start
just allocates the struct and calls reset.

pokey_device::sound_stream_update -> pokey_device_update

Unsure if the other output_types can be used anywhere within LibVGM?
Also unsure if the other output types are correct, or if they should all
just be removed.

Verified everything is static except the devDefList_Pokey

### CODE REVIEW

**Comment by jprjr** [comment]
See also issue #43

File: emu/cores/pokey.c [hunk 1]
```diff
@@ -165,1028 +330,727 @@ const DEV_DEF* devDefList_Pokey[] =
...
+	d->m_out_filter = 0;
+	d->m_out_raw = 0;
+	d->m_old_raw_inval = 1;
+	d->m_kbd_state = 0;
+
+	d->m_icount = 0;
+
+	/* reset more internal state */
+	memset(d->m_clock_cnt,0,sizeof(d->m_clock_cnt));
+	memset(d->m_clock_cnt,0,sizeof(d->m_POTx));
```
Comment (ValleyBell) [inline emu/cores/pokey.c:501]: I think this should clear `d->m_POTx`.

Comment (jprjr) [reply emu/cores/pokey.c]: Oops! Thanks for that catch, fixed in 3ceaf98b3bbbfcd48c8107cb903aa40f267c3e55

**Comment by ValleyBell** [comment]
Aside from the one line above, this looks good to me!

A few small things: (I don't want them fixed, I'm just documenting what I usually do):
- `pokey_device::vol_init` -> `pokey_vol_init` as the `_device` part is usually redundant
- I usually remove the `m_` prefix from member variables, as we have `d->` in C instead.

**Comment by ValleyBell** [comment]
I checked the volume and it's exactly the same as before in the logged WAV file.
The WAV log is not a binary match as the square phases are a bit different, but that's fine. (maybe be different initialization)
The new core even got rid of small pop at sample 0 in MSVC Debug mode.

Muting works fine.

Again, thank you a lot for the effort!

**Pull request merged**